GH-108009: Add clarification of parser and argument defaults in argparse docs (#124154)
python/cpython: Doc/library/argparse.rst

     >>> parser.parse_args(['736'])
     Namespace(bar=42, baz='badger', foo=736)

   Note that defaults can be set at both the parser level using :meth:`set_defaults`

>    and at the argument level using :meth:`add_argument`. If both are provided for the
>    same argument, the order of setting these defaults determines which one takes
>    precedence. The last default set for an argument is the one that is used::

Reviewer: I think we can simplify this quite a bit to make it clearer, more direct and easier to follow, and at least partially address others users' concerns about potentially overcomplicating the docs:

```suggestion
   and at the argument level using :meth:`add_argument`. If both are called for the
   same argument, the last default set for an argument is used::
```

Author: Thanks for the review and the suggestion. This gets the point across in fewer words!
